php/integrations/yoast.php

```diff
@@ -0,0 +1,243 @@
+		\add_filter( 'wpseo_schema_profilepage', [ __CLASS__, 'filter_schema_profilepage' ], 11, 4 );
+		\add_filter( 'wpseo_meta_author', [ __CLASS__, 'filter_author_meta' ], 11, 2 );
+	}
+
+	/**
+	 * Filters the graph output of authors archive for guest authors.
+	 *
+	 * @param array                   $data                   The schema graph.
+	 * @param Meta_Tags_Context       $context                The context object.
+	 * @param Abstract_Schema_Piece   $graph_piece_generator  The graph piece generator.
+	 * @param Abstract_Schema_Piece[] $graph_piece_generators The graph piece generators.
+	 *
+	 * @return array The (potentially altered) schema graph.
+	 */
+	public static function filter_schema_profilepage( $data, $context, $graph_piece_generator, $graph_piece_generators ) {
+
+		if ( ! is_author() ) {
+			return $data;
+		}
+
+		$user = \get_queried_object();
+
+		if ( empty( $user->type ) || $user->type !== 'guest-author' ) {
+			return $data;
+		}
+
+		// Fix author URL.
+		$author_url                                     = \get_author_posts_url( $user->ID, $user->user_nicename );
+		$graph_piece_generator->context->canonical      = $author_url;
+		$graph_piece_generator->context->main_schema_id = $author_url;
+
+		return $graph_piece_generator->generate();
+	}
+
+	/**
+	 * Filters the graph output to add authors.
+	 *
+	 * @param array                   $data                   The schema graph.
+	 * @param Meta_Tags_Context       $context                The context object.
+	 * @param Abstract_Schema_Piece   $graph_piece_generator  The graph piece generator.
+	 * @param Abstract_Schema_Piece[] $graph_piece_generators The graph piece generators.
+	 *
+	 * @return array The (potentially altered) schema graph.
+	 */
+	public static function filter_author_graph( $data, $context, $graph_piece_generator, $graph_piece_generators ) {
+		if ( ! isset( $data['image']['url'] ) ) {
+			return $data;
+		}
+
+		if ( isset( $data['image']['@id'] ) ) {
+			$data['image']['@id'] .= \md5( $data['image']['url'] );
+		}
+
+		if ( isset( $data['logo']['@id'] ) ) {
+			$data['logo']['@id'] .= \md5( $data['image']['url'] );
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Filters the graph output to add authors.
+	 *
+	 * @param array             $data    The schema graph.
+	 * @param Meta_Tags_Context $context Context object.
+	 *
+	 * @return array The (potentially altered) schema graph.
+	 */
+	public static function filter_graph( $data, $context ) {
+		if ( ! \is_singular() ) {
+			return $data;
+		}
+
+		if ( ! \function_exists( '\get_coauthors' ) ) {
+			return $data;
+		}
+
+		/**
+		 * Contains the authors from the CoAuthors Plus plugin.
+		 *
+		 * @var \WP_User[] $author_objects
+		 */
+		$author_objects = \get_coauthors( $context->post->ID );
+
+		$ids     = [];
+		$authors = [];
+
+		// Add the authors to the schema.
+		foreach ( $author_objects as $author ) {
+			$author_generator          = new CoAuthor();
+			$author_generator->context = $context;
```

Comment:
If context and Yoast helpers are required for the `CoAuthor` class, why not add them on the class constructor? Just in case anyone tries to instantiate a `CoAuthors\Integrations\Yoast\CoAuthor` class without these items.

Reply:
Makes sense. I'll have a look. Again something that comes from Yoast and this particular piece not written by me

Reply:
So, the context I only have in this context, so I can't add this into the constructor, unless I make it a required argument.

But this `CoAuthor` class extends Yoasts' `Author` which extends `Person` and `Abstract_Schema_Piece`. I don't want to change the classe structure.

Also, this piece of the code was written by one of the founders of Yoast, so I prefer leaving it as it is :)

Reply:
Sounds good!